### PR TITLE
Add token to the IPN redirect URL in case the user is not authenticated

### PR DESCRIPTION
**Pull request by robinmitra**
The user would see access denied message otherwise.

### CODE REVIEW

**Comment by guanhuan** [comment]
Hi @colemanw ,

Currently when people complete an offsite payment and gets back to the confirmation page, they will see an "access denied" message. We have noticed that for normal onsite submission of the form, an extra token is added in the URL. This PR is to add that back to offsite payment.

Best

Guanhuan

File: includes/wf_crm_webform_postprocess.inc [hunk 1]
```diff
@@ -1705,7 +1705,13 @@ class wf_crm_webform_postprocess extends wf_crm_webform_base {
       $url = url('node/' . $this->node->nid, array('absolute' => TRUE));
     }
     elseif ($url == '<confirmation>') {
-      $url = url("node/{$this->node->nid}/done", array('query' => array('sid' => $this->submission->sid), 'absolute' => TRUE));
+      global $user;
+      $query = array('sid' => $this->submission->sid);
+      // Add token if user is not authenticated, inline with 'webform_client_form_submit()'
+      if ((int) $user->uid === 0) {
```
Comment (colemanw) [inline includes/wf_crm_webform_postprocess.inc]: This line could be simplified as `if (user_is_anonymous())` and the line above (`global $user;`) could be removed.

**Comment by colemanw** [comment]
Thanks @guanhuan for the fix.

**Comment by colemanw** [comment]
@robinmitra see one coding suggestion above.

**Comment by robinmitra** [comment]
Thanks for your comment @colemanw - have updated the PR accordingly.

**Pull request merged**